stdlib/public/core/StaticBigInt.swift

```diff
@@ -0,0 +1,225 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2022 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+/// An *immutable* arbitrary-precision signed integer.
+///
+/// `StaticBigInt` is primarily intended to be used as the associated type of an
+/// ``ExpressibleByIntegerLiteral`` conformance.
+///
+///     extension UInt128: ExpressibleByIntegerLiteral {
+///         public typealias IntegerLiteralType = StaticBigInt
+///         public init(integerLiteral value: IntegerLiteralType) {
+///             precondition(
+///                 value.bitWidth <= 129 && !value.isNegative,
+///                 "integer literal overflows when stored into 'UInt128'"
+///             )
+///             // Copy the elements of `value.words` into this instance.
+///             // Avoid numeric APIs that may trigger infinite recursion.
+///         }
+///     }
+///
+/// `StaticBigInt` can't conform to any numeric protocols, but it does implement
+/// the relational operators (`<`, `<=`, `>=`, `>`, `==`, `!=`) and some numeric
```

Comment:
What do we need the comparisons for? Can they actually be used anywhere, or are we just implementing them because we can?

Reply:
> What do we need the comparisons for?

The `isNegative` property is a "non-standard" API (i.e. not in the numeric protocols). So it was originally internal, and `value < 0` would have been the public API. I changed my mind about that, but the implementations for `<` and `==` are simple enough that I've kept them in the design. I don't know how useful they'll be.

Reply:
> Can they actually be used anywhere, or are we just implementing them because we can?

These APIs were removed in commit 1286e0ab33f39409dd8d2f2f1b3cb02f6447b4c4.